src/main/java/thaumicenergistics/common/container/ContainerDistillationPatternEncoder.java

```diff
@@ -143,27 +134,22 @@ public boolean isItemValidForSlot(final int slotIndex, final ItemStack itemStack
         };
 
         // Add the source aspect slots
-        for (int index = 0; index < ContainerDistillationPatternEncoder.SLOT_SOURCE_ASPECTS_COUNT; ++index) {
+        // Row
+        for (int index = 0; index < 4; ++index) {
             // Calculate Y
             int posY = ContainerDistillationPatternEncoder.SLOT_SOURCE_ASPECTS_POS_Y + (index * 18);
-
-            // Create the slot
-            this.slotSourceAspects[index] = new SlotFake(
-                    this.internalInventory,
-                    index,
-                    ContainerDistillationPatternEncoder.SLOT_SOURCE_ASPECTS_POS_X,
-                    posY);
-            this.addSlotToContainer(this.slotSourceAspects[index]);
+            // Line
+            for (int line = 0; line < 4; line++) {
+                // Create the slot
+                this.slotSourceAspects[index * 4 + line] = new SlotFake(
+                        this.internalInventory,
+                        index * 4 + line,
+                        ContainerDistillationPatternEncoder.SLOT_SOURCE_ASPECTS_POS_X + 18 * line,
+                        posY);
+                this.addSlotToContainer(this.slotSourceAspects[index * 4 + line]);
+            }
         }
```

Comment:
nitpick: Use can use single loop with original constant (`SLOT_SOURCE_ASPECT_COUNT`) and retrieve `line` index using bit manipulation:
```java
int line = index & 0x3;
```

Reply:
I know this trick long time ago, but never used yet, already change it